Formula/re-flex.rb

```diff
@@ -1,23 +1,29 @@
 class ReFlex < Formula
+  env :std
   desc "The regex-centric, fast and flexible scanner generator for C++"
   homepage "https://www.genivia.com/doc/reflex/html"
-  url "https://github.com/Genivia/RE-flex/archive/v1.6.6.tar.gz"
-  sha256 "be98ae3b270ebec4261f5bc78b5bbbe02ce6afee080c175c9a6bb8f01e28eaee"
+  url "https://github.com/Genivia/RE-flex/archive/v1.6.7.tar.gz"
+  sha256 "526e66f7317cfaa9ea017ad0a83a5c2f8ae469f9c522eed989eb83f16342c72d"
 
   bottle do
     cellar :any_skip_relocation
-    sha256 "b3aaeca6a4d5b42f3bd117a63f33000e7997ac2613eb3724b7452c060c05980e" => :catalina
-    sha256 "e5159fd4b0064a475cec8065366482dbcc3335f899bf35446e71fb90cbed305e" => :mojave
-    sha256 "fa43c6264ba478c09aeffd748e5e8242b1ac678d4470d3b12187f9d7ec9f1628" => :high_sierra
+    sha256 "075657453a8dc116d425965a5ad8432b3b66b8b9cfd581eabdcaa0c595e5ebf1" => :catalina
+    sha256 "caf15bf38d93db054e1f90a9182802b7df688b3266dccd48ccd7550c7d7d50e7" => :mojave
+    sha256 "aeee136f831845272ba2e5b67aee5cbe6e93a6d4b77a9991f5bd0bda88f93e76" => :high_sierra
   end
 
-  depends_on "boost"
-  depends_on "pcre2"
+  depends_on "pcre2" => :recommended
+  depends_on "boost" => :optional
 
   def install
+    ENV.O2
+    ENV.deparallelize
+    ENV.delete("CFLAGS")
+    ENV.delete("CXXFLAGS")
```

Comment:
I suspect these are only needed because you are using `env :std`.

Reply:
The reason for using `env :std` is that this happens (as observed when compiling the same source code https://github.com/Genivia/ugrep/issues/21):
~~~
Last 15 lines from /Users/enrico/Library/Logs/Homebrew/ugrep/02.make:
          __m256i vlcpeq = _mm256_cmpeq_epi8(vlcp, vlcpm);
                           ^
matcher.cpp:504:28: error: always_inline function '_mm256_cmpeq_epi8' requires target feature 'avx2', but would be inlined into function 'advance' that is compiled without support for 'avx2'
          __m256i vlcseq = _mm256_cmpeq_epi8(vlcs, vlcsm);
                           ^
matcher.cpp:505:27: error: always_inline function '_mm256_movemask_epi8' requires target feature 'avx2', but would be inlined into function 'advance' that is compiled without support for 'avx2'
          uint32_t mask = _mm256_movemask_epi8(_mm256_and_si256(vlcpeq, vlcseq));
                          ^
matcher.cpp:505:48: error: always_inline function '_mm256_and_si256' requires target feature 'avx2', but would be inlined into function 'advance' that is compiled without support for 'avx2'
          uint32_t mask = _mm256_movemask_epi8(_mm256_and_si256(vlcpeq, vlcseq));
~~~
The reason for this failure is that `superenv` removes `-march=native` to replace with `-march=core2`. That breaks the build to compile with AVX or SSE2, i.e. `./configure` detects `-march=native` are needed to get AVX or SSE2 support.

Reply:
I presume that this is new code in 1.6.7, considering 1.6.6 worked 3 weeks ago and the `march` stuff hasn't been changed in over a year.

It might be possible to do `ENV["HOMEBREW_OPTFLAGS"] = "-march=native"`. However, I do not recommend doing this as using `native` will rely on whatever the CI uses - which may not be constant.

A better option would be doing something like `ENV.append_to_cflags "-mavx2"`, if that works. We have many AVX formulae that ship without any modifications because they use these flags in their configure script.

Reply:
Good point. But in our case `-mavx2` is also added (and not removed) but not sufficient. Both `-march=native` and `-mavx2` are needed, when AVX is detected by `./configure`. This is due to some AVX instructions that cause problems with clang. I could not find any other way to work around this. I took a fair amount of time to investigate and settled on this that works fine with MacOS and Linux, using GCC (v5 and up) and clang (v9).

Some background: Running `brew install` with a local formula does not work and I get the errors shown. But maybe I am wrong about running `brew install` locally versus running `brew install` with a formula in the brew repo. Is there a difference?

Reply:
Note: the `./configure` script checks if `-march=native` works/compiles by compiling snippets of AVX/SSE2 code, see [`configure.ac`](https://github.com/Genivia/RE-flex/blob/master/configure.ac). Therefore, `./configure` makes sure the build should be fine when `-march=native` and `-mavx` are used (or `-msse2` in case of SSE2). Otherwise, we don't get the feature that the HW offers, basically downgrading the actual HW capabilities.

Reply:
Digging a bit further, this [SO post](https://stackoverflow.com/questions/47207637/clang-generates-errors-when-i-try-to-use-avx-features) in the comment section below appears to suggest that clang-ci does not support `-march=native`. Maybe this is what you're hinting at? But that conflicts with my own experience with CI that shows there is no problem, see for example [ugrep clang CI](https://travis-ci.org/github/Genivia/ugrep/jobs/681509116) results with these flags to compile the same source code (and some more).

Reply:
Just note that the hardware that builds the formula is very likely not the hardware that runs the formula. The build is done on Mac Pro's in our CI, the runtime can be any device that is able to run High Sierra or newer.